packages/block-library/src/accordion-item/index.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Server-side rendering of the `core/accordion-item` block.
+ *
+ * @package WordPress
+ * @since 6.6.0
+ *
+ * @param array $attributes The block attributes.
+ * @param string $content The block content.
+ *
+ * @return string Returns the updated markup.
+ */
+function block_core_accordion_item_render( $attributes, $content ) {
+	if ( ! $content ) {
+		return $content;
+	}
+
+	$p         = new WP_HTML_Tag_Processor( $content );
+	$unique_id = wp_unique_id( 'accordion-item-' );
+
+	// If the item is open by default,
+	// add its id to the accordion group's initial context.isOpen list.
+	if ( $attributes['openByDefault'] ) {
+		// phpcs:ignore Gutenberg.CodeAnalysis.ForbiddenFunctionsAndClasses.ForbiddenFunctionCall
+		gutenberg_block_core_accordion_group_item_ids( $unique_id );
```

Comment:
We need to use it in this other block, or find an alternative pattern.

Reply:
I tried an alternative in https://github.com/WordPress/gutenberg/pull/64119/commits/3e7cae6a488daf4eba61b06653696958d11e1eb5 which alters how the value of `state.isOpen` is initialized on the server based on the block attribute (`openByDefault`), and uses a callback to ensure the context matches, avoiding the shared function. What do you think?

Reply:
It's not so clean, but if that works, it's fine 🙂

By the way, I think it was me who suggested using `(bool)`, but I think it's not right; it's better to use the strings `true` or `false` without quotes, so that they are a boolean property in the generated JSON. Something like this:

```php
$open_by_default = $attributes['openByDefault'] ? 'true' : 'false';
$p->set_attribute( 'data-wp-context', '{ "id": "' . $unique_id . '", "openByDefault": ' . $open_by_default . ' }' );
```

Reply:
Addressed in [9ab8033](https://github.com/WordPress/gutenberg/pull/64119/commits/9ab80339c66dc635ffede2785578ff7e5b93e1c8).